src/common/load-script-icon.js

```diff
@@ -0,0 +1,37 @@
+const images = {};
+
+/**
+ * Sets script's safeIcon property after the image is successfully loaded
+ * @param {VMScript} script
+ * @param {?string} [defaultIcon]
+ * @returns {Promise<boolean>}
+ */
+export function loadScriptIcon(script, defaultIcon = null) {
+  const { key = 'safeIcon' } = loadScriptIcon;
+  const { icon } = script.meta;
+  const url = script.custom?.pathMap?.[icon] || icon;
+  const isNewUrl = url !== script[key];
+  let promise = isNewUrl && url ? images[url] : Promise.resolve(false);
+  if (!promise) {
+    promise = Promise.resolve(loadScriptIcon.store.cache[url] || fetchImage(url));
```

Comment:
How about just attaching `cache` so that the structure of `store` does not matter?

Reply:
Done.